crates/biome_yaml_parser/src/token_source.rs

```diff
@@ -23,18 +21,17 @@ impl<'source> YamlTokenSource<'source> {
 
     pub fn from_str(source: &'source str) -> Self {
         let lexer = YamlLexer::from_str(source);
-        let lexer = BufferedLexer::new(lexer);
 
         let mut source = YamlTokenSource::new(lexer);
-        source.next_non_trivia_token(YamlLexContext::Regular, true);
+        source.next_non_trivia_token(true);
         source
     }
 
-    fn next_non_trivia_token(&mut self, context: YamlLexContext, first_token: bool) {
+    fn next_non_trivia_token(&mut self, first_token: bool) {
```

Comment:
Also, why did you remove the lex context?

Reply:
So previously the lexer was stateless, it relied on the parser's instruction in how to lex the next token, which is what we used the context for. Now, the lexer manages its own state and knows whether to lex a mapping token, a sequence token, and so on. The parser no longer need to pass that information down.